Assets/Scripts/PackItUp/MapGenerator/Level.cs

```diff
@@ -0,0 +1,110 @@
+using PackItUp.Controllers;
+using PackItUp.Interactables;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Events;
+using UnityEngine.UIElements;
+
+namespace PackItUp.MapGenerator {
+    public class Level : MonoBehaviour
+    {
+        [field: SerializeField]
+        public AbstractMapGenerator Generator { get; set; }
+
+        [field: SerializeField]
+        public List<Room> Rooms { get; set; }
+
+        [field: SerializeField]
+        public Room StartRoom { get; set; }
+
+        [field: SerializeField]
+        public HashSet<Vector2Int> Floor { get; set; } = new HashSet<Vector2Int>();
+
+        [field: SerializeField]
+        public TopDownCharacterController[] Players { get; set; }
+
+        // Maintains a list of rooms assigned to players, exit, etc
+        private HashSet<int> _usedRoomIndexes = new HashSet<int>();
+
+        [field: SerializeField]
+        public UnityEvent OnResetLevel { get; set; }
+
+        [SerializeField] private EndZone[] _endZones;
+        private List<object> _items;
+
+        public EndZone[] GetEndZones()
+        {
+            return _endZones;
+        }
+
+        private void Awake()
+        {
+            Players = FindObjectsOfType<TopDownCharacterController>();
```

Comment:
Both of these can be retrieved from the GameManager without having to use Unity methods, don't know if that might be faster